pallets/liquidity-pools-gateway/src/message.rs

```diff
@@ -1,21 +1,21 @@
-use cfg_types::domain_address::{Domain, DomainAddress};
+use cfg_types::domain_address::DomainAddress;
 use frame_support::pallet_prelude::{Decode, Encode, MaxEncodedLen, TypeInfo};
 
 /// Message type used by the LP gateway.
 #[derive(Debug, Encode, Decode, Clone, Eq, MaxEncodedLen, PartialEq, TypeInfo)]
-pub enum GatewayMessage<AccountId, Message> {
+pub enum GatewayMessage<AccountId, Message, Hash> {
 	Inbound {
 		domain_address: DomainAddress,
 		message: Message,
 	},
 	Outbound {
 		sender: AccountId,
-		destination: Domain,
 		message: Message,
+		router_hash: Hash,
```

Comment:
Why is this need it?

Reply:
In the `OutboundMessageHandler` implementation we queue one message per router. Later on when the queue is processing messages, we retrieve the router based on its hash - https://github.com/centrifuge/centrifuge-chain/blob/feat/lp-v2-gateway-multi-router/pallets/liquidity-pools-gateway/src/lib.rs#L743

Reply:
Makes sense!